### PR TITLE
Implement vale style with alex and proselint

### LINKED ISSUE #7: Add the Vale styles to this repository including a vale.ini

### PR DESCRIPTION
**Pull request by ngken0995**
## Fixes

Fixes #7  by @sarayourfriend 

## Description

Integrate Vale (https://github.com/errata-ai/vale) to ensure that Openverse documentation has a consistent style guide that can be enforced. The style guide is using alex and proselint. Alex catch insensitive and inconsiderate writing. Proselint is a linter for English prose.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

### CODE REVIEW

File: styles/alex/Press.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+extends: substitution
+message: Consider using '%s' instead of '%s'.
+ignorecase: true
+level: warning
```
Comment (sarayourfriend) [inline styles/alex/Press.yml:5]:
```suggestion
level: error
```

**Review by sarayourfriend — COMMENTED** [review]
Okay, I've spent the last ~two~ three (it took another hour of digging and experimentation after I started writing this comment to finally get to a good suggestion!) hours or so looking at this PR, reading Vale documentation, and getting a better understanding of how this all works. My understanding currently is:

1. [We should not commit the packaged style files to our repository](https://vale.sh/docs/topics/packages/#packages-and-vcs). The reason for this (aside from Vale suggesting it) being evident in the `alex` styles where there's been a fix to fix what looks to be a bad find/replace job that resulted in `tsuggestionist` being included in the words list instead of `terrorist`. Being able to run `vale sync` without needing to _undo_ the changes to the levels, is important.
  * BTW, despite this, I still think it's the right decision to not include vale in the monorepo, because while we won't be _committing_ the files, to use vale, you still need to run sync, which would add the files to the project directory anyway, making it too easy to accidentally open a list of sensitive terms.
3. Vale has two tools for making more specific changes to rule configuration:
  1. In `vale.ini` we can force rules to specific levels by adding a [format specific setting](https://vale.sh/docs/topics/config/#format-specific-settings), for example, to set `proselint.Typography` to error, we'd do:
  ```
  [*]
  proselint.Typography = error
  ```
  That way we do not need to commit the style to our git repository (and therefore have to do some work to review changes to it, which we don't line-by-line review for any other kind of dependency) but can still manage overrides for rule levels.
  2. We can add an [Openverse vocabulary](https://vale.sh/docs/topics/vocab/) to exclude terms we don't want to match from matching.

After doing more and more reading through all the available packages, I am so torn on which to depend on. I think we can safely ditch alex, mostly because the profanity lists are too cumbersome and produce lots of false positives. For example, when I run it against `documentation/project/planning.md` in the monorepo (using it as an example because it is a long document), the alex rules complain about words like "period", "failure", and "execution". This is too tedious to work around. We'd either have to disable the profanity lists altogether or add these terms to our accept vocab list. Disabling the rules would disable most of the "unique" parts of alex (that is, the parts that aren't covered by other rule sets), and adding individual exceptions is tedious and in the end defeats the purpose of relying on a package. We'd write just as many exceptions as there would be terms left to actually check.

For that reason, I think we should not use alex. Your judgement was good here to not use it in the first place!

I think proselint covers most of the most important cases. It covers, broadly, things that make writing difficult to read and understand. I looked at write-good as well, however it's passive voice rule is worse than RedHat's (RedHat's has more leeway and a better explanatory document for when passive voice should be used) and the e-prime rule is borderline useless and suggests increasing sentence complexity (and knowing what "E-Prime" is). I wanted to include RedHat, but when running vale against our documentation with it, there are too many warnings and errors to take care of all in one go, so it's something we should work towards enabling piecemeal (maybe rule-by-rule). I think the vale.ini could look something like this:

```ini
StylesPath = styles

MinAlertLevel = suggestion
Vocab = Openverse
Packages = proselint

[*]
BasedOnStyles = proselint

# Avoid Vale.Spelling for now, but we do want Vale.Terms for the vocab at least
Vale.Terms = YES

# The suggested terms are obscure
proselint.AnimalLabels = NO
```

Running vale with that configuration against our documentation (excluding generated files including _serve, .venv and the changelogs) produced: `✖ 126 errors, 0 warnings and 0 suggestions in 186 files.`. I think that's more than enough for us to start with :sweat_smile:.

So to summarise the requested changes:

* Move vale configuration and styles into its own directory, just to keep things on the top-level a bit cleaner in this repository.
* Add `styles/*` to `.gitignore` but include the vocab:
```.gitignore
styles/*
!styles/Vocab/
```
* Add `styles/Vocab/Openverse/accept.txt` and add the following words to the list:
  * `Openverse`
  * `WordPress`
  * `GitHub` (Vale complains about "Github" but gives no suggested alternative on its own)
* Update `vale.ini` to the suggested one above, applying the rules to all file types and enabling Vale.Terms.

Thanks again for all the work here. Vale is such an interesting tool, but seems to need so much care to configure correctly and to avoid too many false positives at the start.

**Comment by sarayourfriend** [comment]
@ngken0995 Would you like to continue working on this? If not, I'm happy to take it over from you, just let me know. I know it's gotten a whole lot more complex than any of us originally anticipate it would be, so it's totally no worries if it's outside the scope of your interests at this point.

**Comment by ngken0995** [comment]
@sarayourfriend This is interesting to me but there is other tasks I want to complete([#3361](https://github.com/WordPress/openverse/pull/3361) and [#3258](https://github.com/WordPress/openverse/pull/3258)) Please feel free to take it over. It will be great to observe the process of implementing vale.

**Comment by sarayourfriend** [comment]
Great! Thanks for getting back, those other PRs are very cool as well :sunglasses: I'll ping you for a review here when I've got something that might work.

**Comment by sarayourfriend** [comment]
@ngken0995 I'm going to open a separate PR for this, I think, to make it easier to work on for me. Just wanted to give an explanation for why I'm closing it :+1:
